modules/xfeatures2d/src/beblid.cpp

```diff
@@ -281,27 +327,54 @@ void BEBLID_Impl::compute(InputArray _image, vector<KeyPoint> &keypoints, Output
     CV_DbgAssert(descriptors.type() == CV_8UC1);
 
     // Compute the BEBLID descriptors
-    computeBEBLID(integralImg, keypoints, descriptors);
+    computeBoxDiffsDescriptor(integralImg, keypoints, descriptors);
 }
 
 // constructor
-BEBLID_Impl::BEBLID_Impl(float scale_factor, int n_bits)
+template <class WeakLearnerT>
+BEBLID_Impl<WeakLearnerT>::BEBLID_Impl(float scale_factor, int n_bits)
     : scale_factor_(scale_factor), patch_size_(32, 32)
 {
-    #include "beblid.p512.hpp"
-    #include "beblid.p256.hpp"
-    if (n_bits == SIZE_512_BITS)
-        wl_params_.assign(wl_params_512, wl_params_512 + sizeof(wl_params_512) / sizeof(wl_params_512[0]));
-    else if(n_bits == SIZE_256_BITS)
-        wl_params_.assign(wl_params_256, wl_params_256 + sizeof(wl_params_256) / sizeof(wl_params_256[0]));
+    WeakLearnerT * begin_ptr, * end_ptr;
+    if (n_bits == BEBLID::SIZE_512_BITS)
+    {
+        #include "beblid.p512.hpp"
+        begin_ptr = (WeakLearnerT *) std::begin(beblid_wl_params_512);
+        end_ptr = (WeakLearnerT *) std::end(beblid_wl_params_512);
```

Comment:
> `(WeakLearnerT *)`

Using `WeakLearnerT = ABWLParamsFloatTh` and `BEBLID::SIZE_512_BITS)` is not a valid case and we should emit error.

Probably it makes sense to specialize ctor and/or add `wl_params` parameter.

```
template <class WeakLearnerT>
BEBLID_Impl<WeakLearnerT>::BEBLID_Impl(float scale_factor, int n_bits, std::vector<WeakLearnerT>&& wl_params)
```

and move parameters vector initialization on upper (caller) level (`::create()` calls).

---

Also these changes would not require special values like 102,103 for enum values.